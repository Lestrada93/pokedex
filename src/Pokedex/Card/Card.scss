@import "./../../assets/styles/layout";
@import "./../../assets/styles/colors";
$cardSize: 200px;

.card {
  width: $cardSize;
  height: $cardSize;
  border: 1px solid $light-dark;
  margin: $space;
  background-color: $white;
  cursor: pointer;

  &:hover {
    cursor: pointer;
    box-shadow: 0 0 4px rgba(0, 0, 0, 0.5);
  }

  &__name {
    font-size: 20px;
    font-weight: bold;
    margin: 5px;
    text-transform: capitalize;
  }

  &__id {
    font-style: italic;
  }
}
