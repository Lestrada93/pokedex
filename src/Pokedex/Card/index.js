import React from 'react';
import PropTypes from 'prop-types';
import './Card.scss';

/**
 * Card
 * @description Card component to display the pokemon list in the main grid
 * @param {String} id
 * @param {String} name
 * @param {Function} setPokemonId
 * @returns {*}
 * @constructor
 */
function Card({ id, name, setPokemonId }) {
  return (
    <button
      type="button"
      data-qa={`click-open-detail-${name}`}
      className="card"
      onClick={() => setPokemonId(id)}
    >
      <img
        src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`}
        alt="PokeImage"
      />
      <h4 className="card__name">{name}</h4>
      <span className="card__id">{id}</span>
    </button>
  );
}

Card.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string,
  setPokemonId: PropTypes.func,
};

Card.defaultProps = {
  id: '',
  name: '',
  setPokemonId: () => {},
};

export default Card;
