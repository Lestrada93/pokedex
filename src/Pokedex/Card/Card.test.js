import React from 'react';
import { create } from 'react-test-renderer';
import Card from '.';

describe('Card component', () => {
  test('Snapshot With data', () => {
    const card = create(<Card
      id="1"
      name="Pikachu"
      setPokemonId={() => {}}
    />);
    expect(card.toJSON()).toMatchSnapshot();
  });
});
