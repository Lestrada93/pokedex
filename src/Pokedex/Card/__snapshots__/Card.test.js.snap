// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Snapshot With data 1`] = `
<button
  className="card"
  data-qa="click-open-detail-Pikachu"
  onClick={[Function]}
  type="button"
>
  <img
    alt="PokeImage"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
  />
  <h4
    className="card__name"
  >
    Pikachu
  </h4>
  <span
    className="card__id"
  >
    1
  </span>
</button>
`;
